pkg/engine/generation.go

```diff
@@ -81,6 +81,9 @@ func filterRule(rule kyverno.Rule, policyContext *PolicyContext) *response.RuleR
 		return nil
 	}
 
+	policyContext.JSONContext.Checkpoint()
+	defer policyContext.JSONContext.Restore()
```

Comment:
Can you elaborate on why we need to copy/restore context per rule?

Reply:
Each rule can have separate variables in the context. For example, if a rule loads a config map, it is meant to be used as part of processing that rule and should not be available to subsequent rules.

Reply:
Each rule can have separate variables in the context. For example, if a rule loads a config map, it is meant to be used as part of processing that rule and should not be available to subsequent rules.